packages/plugin-commands-setup/src/setup.win32.ts

```diff
@@ -0,0 +1,67 @@
+import execa from 'execa'
+
+type IEnvironmentValueMatch = { groups: { name: string, type: string, data: string } } & RegExpMatchArray
+
+export async function setupEnvironmentPath (pnpmHomeDir: string): Promise<string> {
+  const pathRegex = /^ {4}(?<name>PATH) {4}(?<type>\w+) {4}(?<data>.*)$/gim
+  const pnpmHomeRegex = /^ {4}(?<name>PNPM_HOME) {4}(?<type>\w+) {4}(?<data>.*)$/gim
+  const regKey = 'HKEY_CURRENT_USER\\Environment'
+
+  const queryResult = await execa('reg', ['query', regKey])
+
+  if (queryResult.failed) {
+    return 'Win32 registry environment values could not be retrieved'
+  }
+
+  const queryOutput = queryResult.stdout
+  const pathValueMatch = [...queryOutput.matchAll(pathRegex)] as IEnvironmentValueMatch[]
+  const homeValueMatch = [...queryOutput.matchAll(pnpmHomeRegex)] as IEnvironmentValueMatch[]
+
+  let commitNeeded = false
+  let homeDir = pnpmHomeDir
+  const logger = []
+
+  if (homeValueMatch.length === 1) {
+    homeDir = homeValueMatch[0].groups.data
+    logger.push(`Currently 'PNPM_HOME' is set to '${homeDir}'`)
+  } else {
+    logger.push(`Setting 'PNPM_HOME' to value '${homeDir}'`)
+    const addResult = await execa('reg', ['add', regKey, '/v', 'PNPM_HOME', '/t', 'REG_EXPAND_SZ', '/d', homeDir, '/f'])
+    if (addResult.failed) {
+      logger.push(`\t${addResult.stderr}`)
+    } else {
+      commitNeeded = true
+      logger.push(`\t${addResult.stdout}`)
+    }
+  }
+
+  if (pathValueMatch.length === 1) {
+    const pathData = pathValueMatch[0].groups.data
+    if (pathData == null || pathData.trim() === '') {
+      logger.push('Current PATH is empty. No changes to this environment variable are applied')
+    } else {
+      const pathDataUpperCase = pathData.toUpperCase()
+      const homeDirUpperCase = `${homeDir.toUpperCase()};`
+      if (pathDataUpperCase.includes(homeDirUpperCase)) {
+        logger.push('PATH already contains PNPM_HOME')
+      } else {
+        logger.push('Updating PATH')
+        const addResult = await execa('reg', ['add', regKey, '/v', pathValueMatch[0].groups.name, '/t', 'REG_EXPAND_SZ', '/d', `${pathData}${homeDir};`, '/f'])
```

Comment:
PATH in Windows can not usually be expected to end with `;` as far as I know.
Placing PNPM_HOME first in PATH would be great.

```suggestion
        const addResult = await execa('reg', ['add', regKey, '/v', pathValueMatch[0].groups.name, '/t', 'REG_EXPAND_SZ', '/d', `${homeDir}${path.delimiter}${pathData}`, '/f'])
```

Reply:
> Placing PNPM_HOME first in PATH would be great.

yes, it should go first.

Reply:
> I suggest the regedit npm package.
> https://www.npmjs.com/package/regedit

I don't think we need additional dependency just for this simple registry manipulation. As far as I can tell, that package is quite big and all that extra bytes will be used couple of times only.